enhancements/node-tuning/containerize-tuned.md

```diff
@@ -0,0 +1,260 @@
+* It was mentioned that we may need to reduce the TuneD footprint in order to add it to RHCOS - what would that entail?
+
+This wouldn't resolve the issues of TuneD restarts - we'd still need to either allow TuneD to run in one-shot mode or we'd need to allow it to restart safely.
+
+To support running TuneD as a systemd service, we would need changes in NTO to generate the TuneD configuration files and make them available to the TuneD service. A restart would be required to apply the configuration after it was generated by NTO.
+
+#### Do not allow TuneD to do restarts (run one shot)  -  do not reapply
+
+* There were two reasons mentioned why we currently run TuneD in daemon mode (instead of allowing it to apply the setting and exit):
+  * [rtentsk plugin](https://github.com/redhat-performance/tuned/blob/master/tuned/plugins/plugin_rtentsk.py): keeps a socket open to avoid static key IPIs. If TuneD didn't do this we'd need another process to do it. In the past, there was such a binary, called [rt-entsk and it was part of the rt-setup package](https://www.redhat.com/sysadmin/real-time-kernel). It was removed in favor of the rtentsk_plugin.
+  * [cpu plugin](https://github.com/redhat-performance/tuned/blob/bcfbd5de1163f95deb45b5a8319aff99020ccef9/tuned/plugins/plugin_cpu.py#L497): keeps the /dev/cpu_dma_latency file open to control the maximum c-state. It may be possible to eliminate this by disabling specific c-states for each CPU (i.e. writing to /sys/devices/system/cpu/cpuX/cpuidle/stateX/disable) for each unwanted c-state.
+
+* Even if we eliminated the above and allowed TuneD to just run, apply settings and then exit, we'd still have an issue if the TuneD pod restarted. In that case we would need a way to determine that the configuration had already been applied and avoid doing it again.
+
+* One drawback to preventing TuneD from re-applying configuration on a TuneD restart would be that there would be no way to make any kind of (TuneD controlled) configuration change without a full node restart. Do we rely on this behavior in OCP?
+
+#### Taint nodes via NTO
+
+Use a taint to prevent application pods from running before the TuneD pod has applied the configuration (and then NTO would remove the taint and unblock the application pods). There are some concerns with this approach:
+
+* Platform pods would require this taint to be tolerated, possibly requiring changes to a lot of pod specs.
+* How would this work when the node was rebooted and the pods are already scheduled?
+
+#### Layered approach
+
+This would involve adding a layer to RHCOS to add TuneD. Moving TuneD into RHCOS would be preferable.
+
+Potential issues:
+* we had issues having multi layers in the past
+* can cause issues with upgrades
+
+#### Pod ordering problems in general
+
+Alternative ways that pod startup could be ordered - potentially with some upstream k8s changes:
```

Comment:
more than ordering, it seems the problem is that there is no concept of dependencies across pods. May be worth to be spelled out explicitely.